client/extensions/woocommerce/app/order/order-fulfillment/index.js

```diff
@@ -273,13 +276,16 @@ class OrderFulfillment extends Component {
 
 export default connect(
 	( state, { order, site } ) => {
-		const labelsLoaded = wcsEnabled && Boolean( areLabelsLoaded( state, order.id, site.ID ) );
+		const labelsLoaded = wcsEnabled && Boolean( areLabelsFullyLoaded( state, order.id, site.ID ) );
 		const hasLabelsPaymentMethod =
-			wcsEnabled && labelsLoaded && getSelectedPaymentMethod( state, order.id, site.ID );
+			wcsEnabled &&
+			labelsLoaded &&
+			areLabelSettingsPristine( state, site.ID ) &&
```

Comment:
This is a bad UX. If the user leaves the `Settings` screen without saving, he won't be able to purchase a label.

Should we store in the `settings` state the "previous" form values so they're returned if the form is "not pristine"?